src/http/fetch_request.ts

```diff
@@ -133,8 +133,15 @@ export class FetchRequest {
 }
 
 function mergeFormDataEntries(url: URL, entries: [string, FormDataEntryValue][]): URL {
+  const currentUrl = new URL(window.location.href)
+
   for (const [ name, value ] of entries) {
-    url.searchParams.append(name, value.toString())
+    if (currentUrl.searchParams.has(name)) {
+      currentUrl.searchParams.delete(name)
+      url.searchParams.set(name, value.toString())
+    } else {
+      url.searchParams.append(name, value.toString())
```

Comment:
@sstephenson when `value` is a [File](https://developer.mozilla.org/en-US/docs/Web/API/File), will coercing to a String break uploads?

Reply:
Would we be better off guarding these calls with `!(value instanceof File)`?

Reply:
What happens when you submit an `<input type="file">` in a `<form method="get">` without Turbo? The browser just skips the field, I think?

Reply:
I don't think there's anything in FetchRequest's current form that restricts this to idempotent `method="get"` requests. Am I missing something?

Reply:
For what it's worth, this loop is to mutate the URL it's submitted to, and not necessarily the body we're submitting. Would omitting Files entirely avoid the problem of turning a file into a query parameter?